ibis/bigquery/api.py

```diff
@@ -22,42 +27,65 @@
 
 
 def compile(expr, params=None):
-    """
-    Force compilation of expression as though it were an expression depending
-    on BigQuery. Note you can also call expr.compile()
+    """Compile an expression for BigQuery.
 
     Returns
     -------
-    compiled : string
+    compiled : str
+
+    See Also
+    --------
+    ibis.expr.types.Expr.compile
+
     """
     from ibis.bigquery.compiler import to_sql
     return to_sql(expr, dialect.make_context(params=params))
 
 
 def verify(expr, params=None):
-    """
-    Determine if expression can be successfully translated to execute on
-    BigQuery
-    """
+    """Check if an expression can be compiled using BigQuery."""
     try:
         compile(expr, params=params)
         return True
     except com.TranslationError:
         return False
 
 
-def connect(project_id, dataset_id, credentials=None):
-    """Create a BigQueryClient for use with Ibis
+SCOPES = ["https://www.googleapis.com/auth/bigquery"]
+CLIENT_ID = (
+    "546535678771-gvffde27nd83kfl6qbrnletqvkdmsese.apps.googleusercontent.com"
+)
+CLIENT_SECRET = "iU5ohAF2qcqrujegE3hQ1cPt"
+
+
+def connect(
+    project_id: Optional[str] = None,
+    dataset_id: Optional[str] = None,
+    credentials: Optional[google.auth.credentials.Credentials] = None,
+) -> BigQueryClient:
+    """Create a BigQueryClient for use with Ibis.
 
     Parameters
     ----------
-    project_id: str
-    dataset_id: str
-    credentials : google.auth.credentials.Credentials, optional, default None
+    project_id : str
+        A BigQuery project id.
+    dataset_id : str
+        A dataset id that lives inside of the project indicated by
+        `project_id`.
+    credentials : google.auth.credentials.Credentials
 
     Returns
     -------
     BigQueryClient
+
     """
+    if credentials is None:
+        credentials, project_id = pydata_google_auth.default(
+            SCOPES,
+            client_id=CLIENT_ID,
+            client_secret=CLIENT_SECRET,
+        )
```

Comment:
You should probably also override the `credentials_cache` argument so that `pydata_google_auth` used directly saves to a different cache than Ibis. See: https://github.com/pydata/pandas-gbq/blob/547812eaf46f4c8a440d8ae0fdde3af8cd81fd2c/pandas_gbq/auth.py#L106

Reply:
Note: You don't have to change both the filename and the dirname, just one would be sufficient. I just did both in `pandas-gbq` to reuse the same credentials we were before.

Reply:
```
credentials_cache=pydata_google_auth.cache.ReadWriteCredentialsCache(filename="ibis.json"),
```

Reply:
Just out of curiosity where does `ibis.json` live?

Reply:
Ah, nevermind. Found it :)

Reply:
Added `credentials_cache`.